go-controller/pkg/allocator/pod/pod_annotation_test.go

```diff
@@ -235,6 +235,46 @@ func Test_allocatePodAnnotationWithRollback(t *testing.T) {
 			wantReleasedIPsOnRollback: ovntest.MustParseIPNets("192.168.0.3/24"),
 			role:                      types.NetworkRolePrimary,
 		},
+		{
+			name:                   "expect new IP and ipv6 gateway LLA for primary udn layer2 with dual stack",
+			ipam:                   true,
+			idAllocation:           true,
+			persistentIPAllocation: true,
+			args: args{
+				ipAllocator: &ipAllocatorStub{
+					netxtIPs: ovntest.MustParseIPNets("192.168.0.3/24", "2010:100:200::3/60"),
+				},
+				idAllocator: &idAllocatorStub{
+					nextID: 100,
+				},
+			},
+			wantUpdatedPod: true,
+			wantPodAnnotation: &util.PodAnnotation{
+				IPs:            ovntest.MustParseIPNets("192.168.0.3/24", "2010:100:200::3/60"),
+				MAC:            util.IPAddrToHWAddr(ovntest.MustParseIPNets("192.168.0.3/24")[0].IP),
+				Gateways:       []net.IP{ovntest.MustParseIP("192.168.0.1").To4(), ovntest.MustParseIP("2010:100:200::1")},
+				IPv6LLAGateway: util.HWAddrToIPv6LLA(util.IPAddrToHWAddr(ovntest.MustParseIP("100.65.0.4"))),
```

Comment:
This is how it is for the LRP mac address generation it just take [0] that is ipv4 for dual stack.

Reply:
Ah, thanks for the clarification. Would it be worth it to add a test for single stack ipv6?

Reply:
Maybe the proper thing is to read the mac address from the LRP directly instead of re-calculate, but it means we have to traverse more things between all the functions and affect the unit tests and the like, but looks like the proper thing to do.

Reply:
imho if possible please lets stick to the current (wrt to the 2 points)
we are going to drop it once the proper fix is out

Reply:
fair enough, @oshoval Let's add a test for single stack ipv6.

Reply:
@oshoval seems you also need to add a unit test here.